apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/RestResourceBase.java

```diff
@@ -41,10 +41,16 @@ public abstract class RestResourceBase {
   protected static final int DEFAULT_PAGE_SIZE = 100;
 
   private static final Function<String, Uni<? extends Schema.CqlKeyspaceDescribe>>
-      MISSING_KEYSPACE = ks -> Uni.createFrom().nullItem();
+      MISSING_KEYSPACE =
+          ks ->
+              Uni.createFrom()
+                  .failure(
+                      new WebApplicationException(
+                          String.format("Keyspace '%s' not found", ks),
+                          Response.Status.BAD_REQUEST));
 
   private static final Function<String, Uni<? extends QueryOuterClass.Response>>
-      MISSING_KEYSPACE_RESPONSE =
+      MISSING_KEYSPACE_AS_RESPONSE =
```

Comment:
where is this used?

Reply:
aha this is typing problem, uf so bad..

Reply:
yes, exactly. Was trying to think of whether to just cast it (since type not really needed) but decided to keep separate for now.